### ISSUE TITLE
Issue installing `libssl1.1` in section `How to Build NGINX from Source`

### ISSUE THREAD

**Issue by FranciscoRMRZ** [fhsinchy/the-docker-handbook#30]
At the moment the command `docker image build --tag custom-nginx:built .` is run, the following errors show:

```bash
❯ docker image build --tag custom-nginx:built .
[+] Building 6.0s (6/10)
 => [internal] load build definition from Dockerfile                                                   0.0s
 => => transferring dockerfile: 899B                                                                   0.0s
 => [internal] load .dockerignore                                                                      0.0s
 => => transferring context: 2B                                                                        0.0s
 => [internal] load metadata for docker.io/library/ubuntu:latest                                       1.6s
 => CACHED [1/6] FROM docker.io/library/ubuntu:latest@sha256:67211c14fa74f070d27cc59d69a7fa9aeff8e28e  0.0s
 => => resolve docker.io/library/ubuntu:latest@sha256:67211c14fa74f070d27cc59d69a7fa9aeff8e28ea118ef3  0.0s
 => [internal] load build context                                                                      0.0s
 => => transferring context: 1.05MB                                                                    0.0s
 => ERROR [2/6] RUN apt-get update &&     apt-get install build-essential                    libpcre3  4.3s
------
 > [2/6] RUN apt-get update &&     apt-get install build-essential                    libpcre3                     libpcre3-dev                     zlib1g                     zlib1g-dev                     libssl1.1                     libssl-dev                     -y &&     apt-get clean && rm -rf /var/lib/apt/lists/*:
#5 0.438 Get:1 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
#5 0.568 Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
#5 0.813 Get:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
#5 0.905 Get:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [108 kB]
#5 0.971 Get:5 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [917 kB]
#5 0.995 Get:6 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1792 kB]
#5 1.214 Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
#5 1.284 Get:8 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [909 kB]
#5 1.331 Get:9 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [23.2 kB]
#5 1.332 Get:10 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [914 kB]
#5 1.683 Get:11 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]
#5 1.685 Get:12 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]
#5 1.688 Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [1258 kB]
#5 1.737 Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1151 kB]
#5 1.753 Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [28.6 kB]
#5 1.754 Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [967 kB]
#5 1.908 Get:17 http://archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [49.4 kB]
#5 2.131 Get:18 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [23.3 kB]
#5 2.528 Fetched 26.5 MB in 2s (11.7 MB/s)
#5 2.528 Reading package lists...
#5 3.279 Reading package lists...
#5 3.961 Building dependency tree...
#5 4.123 Reading state information...
#5 4.184 E: Unable to locate package libssl1.1
#5 4.184 E: Couldn't find any package by glob 'libssl1.1'
#5 4.184 E: Couldn't find any package by regex 'libssl1.1'
------
executor failed running [/bin/sh -c apt-get update &&     apt-get install build-essential                    libpcre3                     libpcre3-dev                     zlib1g                     zlib1g-dev                     libssl1.1                     libssl-dev                     -y &&     apt-get clean && rm -rf /var/lib/apt/lists/*]: exit code: 100
```
The root cause is the Ubuntu version (Jammy Jellyfish).

Course of action may be one of this (from my point of view):
- Change the `nginx` version and update the dependency list.
- Lock the Ubuntu version to the one used when the book's current version was written.

In the mean time, if I come up with a solution I'll make a pull request.

Best regards

**Comment by FranciscoRMRZ** [comment]
I already sent PR #31 to fix the issue.

If there is anything I can do to help you further, let me know.

Best regards.

**Comment by FranciscoRMRZ** [comment]
Closed by PR #31
